--- src/3rdparty/chromium/base/process/process_metrics_mac.cc	2020-11-06 20:22:36.000000000 -0500
+++ src/3rdparty/chromium/base/process/process_metrics_mac.cc	2020-11-23 12:35:14.000000000 -0500
@@ -293,17 +293,8 @@
                         &count) != KERN_SUCCESS) {
     return false;
   }
-  DCHECK_EQ(HOST_VM_INFO64_COUNT, count);
-
-  static_assert(PAGE_SIZE % 1024 == 0, "Invalid page size");
-  meminfo->free = saturated_cast<int>(
-      PAGE_SIZE / 1024 * (vm_info.free_count - vm_info.speculative_count));
-  meminfo->speculative =
-      saturated_cast<int>(PAGE_SIZE / 1024 * vm_info.speculative_count);
-  meminfo->file_backed =
-      saturated_cast<int>(PAGE_SIZE / 1024 * vm_info.external_page_count);
-  meminfo->purgeable =
-      saturated_cast<int>(PAGE_SIZE / 1024 * vm_info.purgeable_count);
+  meminfo->free = static_cast<int>(
+  (vm_info.free_count - vm_info.speculative_count) * PAGE_SIZE / 1024);
 
   return true;
 }
